portal-web/docroot/html/common/themes/top_messages.jsp

```diff
@@ -27,32 +27,39 @@ String jspPath = (String)PortalMessages.get(request, PortalMessages.KEY_JSP_PATH
 String message = (String)PortalMessages.get(request, PortalMessages.KEY_MESSAGE);
 
 if (Validator.isNotNull(jspPath) || Validator.isNotNull(message)) {
-	String cssClass = GetterUtil.getString(PortalMessages.get(request, PortalMessages.KEY_CSS_CLASS), "alert-info");
+	String cssClass = GetterUtil.getString(PortalMessages.get(request, PortalMessages.KEY_CSS_CLASS), "");
+	String displayType = GetterUtil.getString(PortalMessages.get(request, PortalMessages.KEY_DISPLAY_TYPE), "info");
 	String portletId = (String)PortalMessages.get(request, PortalMessages.KEY_PORTLET_ID);
 	int timeout = GetterUtil.getInteger(PortalMessages.get(request, PortalMessages.KEY_TIMEOUT), 10000);
 %>
 
+	<liferay-util:buffer
+		var="alertMessage"
+	>
+		<c:choose>
+			<c:when test="<%= Validator.isNotNull(jspPath) %>">
+				<liferay-util:include page="<%= jspPath %>" portletId="<%= portletId %>" />
+			</c:when>
+			<c:otherwise>
+				<liferay-ui:message key="<%= message %>" />
+			</c:otherwise>
+		</c:choose>
+	</liferay-util:buffer>
+
 	<aui:script>
 		Liferay.Util.openToast({
-			message: "
-				<c:choose>
-					<c:when test="<%= Validator.isNotNull(jspPath) %>">
-						<liferay-util:include page="<%= jspPath %>" portletId="<%= portletId %>" />
-					</c:when>
-					<c:otherwise>
-						<liferay-ui:message key="<%= message %>" />
-					</c:otherwise>
-				</c:choose>",
+			message: '<%= HtmlUtil.escapeJS(alertMessage) %>',
```

Comment:
Not super fan about this, but maybe we could do here:

```jsp
<c:if test="<%= Validator.isNotNull(onCloseJspPath) %>">
onClose: function(event) {
    <liferay-util.include page="<%= onCloseJspPath %>" porltetId="<%= portletId %>" />
}
</c:if>
```

Reply:
So, inside this onCloseJSP We'll have just a script calling the things that needs to be called when closing, right?